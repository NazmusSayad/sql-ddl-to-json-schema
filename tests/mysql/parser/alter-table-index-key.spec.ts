import { getParsedFormat } from '../parse-handler';
import { run } from '../runner';

run(getParsedFormat, {
  'Parser: Should alter table adding index key with two columns and options.': {
    queries: [
      `ALTER TABLE people add index ik_id using hash ( id(2) asc, o_id ) key_block_size 1024 comment 'test';`,
    ],
  },

  'Parser: Should alter table adding index key with one column and option.': {
    queries: [`ALTER TABLE people add index ik_id(id)key_block_size 1024;`],
  },

  'Parser: Should alter table adding index key with one column.': {
    queries: [`ALTER TABLE people add index ik_id (id);`],
  },

  'Parser: Should alter table adding unnamed index key.': {
    queries: [`ALTER TABLE people add index (id);`],
  },

  'Parser: Should alter table adding key with two columns and options.': {
    queries: [
      `ALTER TABLE people add key ik_id using hash ( id(2) asc, o_id ) key_block_size 1024 comment 'test';`,
    ],
  },

  'Parser: Should alter table adding key with one column and option.': {
    queries: [`ALTER TABLE people add key ik_id(id)key_block_size 1024;`],
  },

  'Parser: Should alter table adding key with one column.': {
    queries: [`ALTER TABLE people add key ik_id (id);`],
  },

  'Parser: Should alter table adding unnamed key.': {
    queries: [`ALTER TABLE people add key (id);`],
  },

  'Parser: Should alter table dropping index.': {
    queries: [`ALTER TABLE people drop index i_oid;`],
  },

  'Parser: Should alter table dropping key.': {
    queries: [`ALTER TABLE people drop key k_oid;`],
  },

  'Parser: Should alter table renaming index.': {
    queries: [`ALTER TABLE people rename index i_oid to ii_oid;`],
  },

  'Parser: Should alter table renaming key.': {
    queries: [`ALTER TABLE people rename key k_oid to kk_oid;`],
  },
});
