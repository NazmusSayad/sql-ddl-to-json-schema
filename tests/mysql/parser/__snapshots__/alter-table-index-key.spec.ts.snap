// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parser: Should alter table adding index key with one column and option. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addIndex",
                    "name": "ik_id",
                    "index": null,
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id"
                        }
                      }
                    ],
                    "options": [
                      {
                        "id": "O_INDEX_OPTION",
                        "def": {
                          "keyBlockSize": 1024
                        }
                      }
                    ]
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding index key with one column. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addIndex",
                    "name": "ik_id",
                    "index": null,
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id"
                        }
                      }
                    ],
                    "options": []
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding index key with two columns and options. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addIndex",
                    "name": "ik_id",
                    "index": {
                      "id": "P_INDEX_TYPE",
                      "def": "hash"
                    },
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id",
                          "length": 2,
                          "sort": "asc"
                        }
                      },
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "o_id"
                        }
                      }
                    ],
                    "options": [
                      {
                        "id": "O_INDEX_OPTION",
                        "def": {
                          "keyBlockSize": 1024
                        }
                      },
                      {
                        "id": "O_INDEX_OPTION",
                        "def": {
                          "comment": "test"
                        }
                      }
                    ]
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding key with one column and option. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addIndex",
                    "name": "ik_id",
                    "index": null,
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id"
                        }
                      }
                    ],
                    "options": [
                      {
                        "id": "O_INDEX_OPTION",
                        "def": {
                          "keyBlockSize": 1024
                        }
                      }
                    ]
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding key with one column. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addIndex",
                    "name": "ik_id",
                    "index": null,
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id"
                        }
                      }
                    ],
                    "options": []
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding key with two columns and options. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addIndex",
                    "name": "ik_id",
                    "index": {
                      "id": "P_INDEX_TYPE",
                      "def": "hash"
                    },
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id",
                          "length": 2,
                          "sort": "asc"
                        }
                      },
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "o_id"
                        }
                      }
                    ],
                    "options": [
                      {
                        "id": "O_INDEX_OPTION",
                        "def": {
                          "keyBlockSize": 1024
                        }
                      },
                      {
                        "id": "O_INDEX_OPTION",
                        "def": {
                          "comment": "test"
                        }
                      }
                    ]
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding unnamed index key. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addIndex",
                    "name": null,
                    "index": null,
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id"
                        }
                      }
                    ],
                    "options": []
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding unnamed key. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addIndex",
                    "name": null,
                    "index": null,
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id"
                        }
                      }
                    ],
                    "options": []
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table dropping index. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "dropIndex",
                    "index": "i_oid"
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table dropping key. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "dropIndex",
                    "index": "k_oid"
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table renaming index. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "renameIndex",
                    "index": "i_oid",
                    "newName": "ii_oid"
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table renaming key. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "renameIndex",
                    "index": "k_oid",
                    "newName": "kk_oid"
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;
